test/FunctionsV2/PlatformSpecificHelpers.FunctionsV2.cs

```diff
@@ -78,6 +78,9 @@ private static IWebJobsBuilder AddDurableTask(this IWebJobsBuilder builder, IOpt
                 case TestHelpers.AzureStorageProviderType:
                     // This provider is built into the default AddDurableTask() call below.
                     break;
+                case TestHelpers.AzureStorageProviderModifiedType:
```

Comment:
I think you could make this a bit more generalized. Instead of adding a new string type, have the `GetJobHost` method take a type of `IDurabilityProviderFactory`, and if that is provided, use that instead of the type specified in the string `storageProvider`. 

That way, you could remove some of the special logic around `"azure_storage_modified"`, and your tests would look like:

```csharp
            using (ITestHost host = TestHelpers.GetJobHost(
                this.loggerProvider,
                nameof(this.LongRunningTimer),
                extendedSessions,
                storageProviderType: "azure_storage",
                storageProviderFactory = typeof(AzureStorageShortenedMaxTimerDurabilityProviderFactory))
```